src/Symfony/Component/Cache/Adapter/ChainAdapter.php

```diff
@@ -61,11 +61,11 @@ public function __construct(array $adapters, int $defaultLifetime = 0)
         $this->adapterCount = \count($this->adapters);
 
         $this->syncItem = \Closure::bind(
-            function ($sourceItem, $item) use ($defaultLifetime) {
+            function ($sourceItem, $sourceMetadata, $item) use ($defaultLifetime) {
                 $item->value = $sourceItem->value;
-                $item->expiry = $sourceItem->expiry;
+                $item->expiry = $sourceMetadata[CacheItem::METADATA_EXPIRY] ?? null;
                 $item->isHit = $sourceItem->isHit;
-                $item->metadata = $sourceItem->metadata;
```

Comment:
I would have preferred to keep reading from $sourceItem->metadata here, but this is cleared by https://github.com/symfony/symfony/blob/7b0ca3ab5678232f876a95991b66778f4aaa33e3/src/Symfony/Component/Cache/Traits/ContractsTrait.php#L65 when the adapter is a `CacheInterface` so have provided the metadata separately to work around this.

Reply:
this is cleared only if the expiry is over, so not sure about this argument

Reply:
That didn't appear to be the case in my testing. The condition makes sense when setting metadata based on the item properties set by the cache interface build callback, however in the case of the chain adapter the wrapper callback will be returning an item fetched from the cache which does not have the expiry property set (see #34487). Thus the metadata is cleared.